pastis/launchers/run_luvoir_efields.py

```diff
@@ -1,58 +1,71 @@
-from pastis.pastis_analysis import  calculate_segment_constraints
-from pastis.simulators.luvoir_imaging import LuvoirA_APLC
+from pastis.pastis_analysis import calculate_segment_constraints
 import pastis.plotting as ppl
 
+
 if __name__ == '__main__':
+
+    NUM_RINGS = 1
+    WHICH_DM = 'harris_seg_mirror'   # 'harris_seg_mirror' or 'seg_mirror', or (global) 'zernike_mirror'
+
     # DM_SPEC = tuple or int, specification for the used DM -
     # for seg_mirror: int, number of local Zernike modes on each segment
     # for harris_seg_mirror: tuple (string, array, bool, bool, bool),
     # absolute path to Harris spreadsheet, pad orientations, choice of Harris mode sets (thermal, mechanical, other)
     # for zernike_mirror: int, number of global Zernikes
 
-    # If Harris deformable mirror, uncomment the following lines
-    # DM = 'harris_seg_mirror'  # Possible: "seg_mirror", "harris_seg_mirror", "zernike_mirror"
-    # fpath = CONFIG_PASTIS.get('LUVOIR', 'harris_data_path')  # path to Harris spreadsheet
-    # pad_orientations = np.pi / 2 * np.ones(CONFIG_PASTIS.getint('LUVOIR', 'nb_subapertures'))
-    # DM_SPEC = (fpath, pad_orientations, True, False, False)
+    # If using Harris deformable mirror
+    if WHICH_DM == 'harris_seg_mirror':
+        fpath = CONFIG_PASTIS.get('LUVOIR', 'harris_data_path')  # path to Harris spreadsheet
+        pad_orientations = np.pi / 2 * np.ones(CONFIG_PASTIS.getint('LUVOIR', 'nb_subapertures'))
+        DM_SPEC = (fpath, pad_orientations, True, False, False)
 
-    # If Segmented Zernike Mirror, uncomment the following lines
-    DM = 'seg_mirror'  # Possible: "seg_mirror", "harris_seg_mirror", "zernike_mirror"
-    DM_SPEC = 5
+    # If using Segmented Zernike Mirror
+    if WHICH_DM == 'seg_mirror':
+        DM_SPEC = 3
 
     APLC_DESIGN = 'small'
     # First generate a couple of matrices
-    run_matrix = MatrixEfieldLuvoirA(which_dm=DM, dm_spec=DM_SPEC, design=APLC_DESIGN,
+    run_matrix = MatrixEfieldLuvoirA(which_dm=WHICH_DM, dm_spec=DM_SPEC, design=APLC_DESIGN,
                                      calc_science=True, calc_wfs=True,
                                      initial_path=CONFIG_PASTIS.get('local', 'local_data_path'),
                                      norm_one_photon=True)
     run_matrix.calc()
     dir_run = run_matrix.overall_dir
     print(f'All saved to {dir_run}.')
 
-    # Calculate and save per segment static tolerance
+    # get the automatically saved pastis_matrix
     pastis_matrix = fits.getdata(os.path.join(dir_run, 'matrix_numerical', 'pastis_matrix.fits'))
 
     # get the unaberrated coro_psf after the matrix run
     e0_psf = fits.getdata(os.path.join(dir_run, 'unaberrated_coro_psf.fits'))  # already normalized to max of direct pdf
     dh_mask = np.array(run_matrix.simulator.dh_mask.shaped)
     contrast_floor = util.dh_mean(e0_psf, dh_mask)
 
-    # set the target contrast
+    # Calculate the static tolerances
```

Comment:
I will still prefer the comment "Set the target contrast", because here want to compute tolerances for a set contrast.